src/sources/lolalytics.js

```diff
@@ -0,0 +1,304 @@
+  const blocks = [];
+  const item_names = store.get('item_names');
+  const sortByWinRate = R.sortBy(R.prop('win'));
+  const sortByPickRate = R.sortBy(R.prop('pick'));
+  const build_sections = parsed_data.build_sections;
+  const skills = parsed_data.skills;
+
+  const build_sections_to_convert = [];
+
+  R.forEach(function(x) { if (R.contains(x.title, item_block_names)) build_sections_to_convert.push(x); }, build_sections); //  Filter out the rows we don't care about (champion counters, etc.)
+
+  function convertBuildSectionJsonToBlock(build_section_json, title) {
+    const items = [];
+
+    for (let k = 0; k < build_section_json.length; k++) {
+      let id = parseInt(item_names[build_section_json[k].name], 10);
+
+      id = (isNaN(id) ? 2003 : id); //  For biscuit
+
+      items.push({
+        count: 1,
+        id: id.toString()
+      });
+    }
+
+    const filtered_items = R.uniqBy(R.prop('id'), items); //  filter out duplicates, mainly just the second health pot from converting buscuits
+
+    let block = {
+      type: title,
+      items: filtered_items
+    };
+
+    return block;
+  }
+
+
+  //  Iterate the sections
+  for (let i = 0; i < build_sections_to_convert.length; i++) {
+    //  Iterate the columns in that section
+    const current_section = build_sections_to_convert[i];
+
+    if (!split_sort) {
+      let block = convertBuildSectionJsonToBlock(R.reverse(sortByWinRate(current_section.json)), `${T.t('highest_win', true)} ${T.t(current_section.title.split(' ').join('_'))}`);
+      blocks.push(block);
+      block = convertBuildSectionJsonToBlock(R.reverse(sortByPickRate(current_section.json)), `${T.t('most_freq', true)} ${T.t(current_section.title.split(' ').join('_'))}`);
+      blocks.push(block);
+    } else if (R.toLower(split_sort) === 'win') {
+      let block = convertBuildSectionJsonToBlock(R.reverse(sortByWinRate(current_section.json)), `${T.t(current_section.title.split(' ').join('_'))}`);
+      blocks.push(block);
+    } else {
+      let block = convertBuildSectionJsonToBlock(R.reverse(sortByPickRate(current_section.json)), `${T.t(current_section.title.split(' ').join('_'))}`);
+      blocks.push(block);
+    }
+  }
+
+  let riot_json = R.merge(R.clone(default_schema, true), {
+    champion: champion_name,
+    title: `Lolalytics ${champion_name} ${T.t(R.toLower(position), true)} ${store.get('lolalytics_ver')}`,
+    blocks: trinksCon(blocks, skills)
+  });
+
+  if (split_sort) {
+    if (R.toLower(split_sort) === 'win') {
+      riot_json.title += ' ' + T.t('highest_win', true);
+    } else {
+      riot_json.title += ' ' + T.t('most_freq', true);
+    }
+  }
+
+  if (store.get('settings').locksr) riot_json.map = 'SR';
+
+  return createChampionifyJson(riot_json, position, champion_name, split_sort);
+}
+
+function parseSkillsData(cheerio_data) {
+  let $c = cheerio_data;
+  let skill_sections = [];
+  $c('td.summarytitle').each(function(i, element) {
```

Comment:
Cheerio has a map option, combining it with `R.flatten` you can avoid having to push to an array. Also if you reassign the element your mapping over, it should be a bit more performance then having cheerio parse it each time. Here's something quickly written, I haven't tested it but should work.

```javascript
let skill_sections = $c('td.summarytitle').map((i, el) => {
  el = $c(el);
  if (el.text() === 'Abilities') {
    return el.parent().find('table.summaryskillstable').find('tr').map((idx, row) => {
      row = $c(row);
      if (idx === 0) return; //  Skip the header row

      const skill_order = row.find('td.summaryskills > div.summaryspellkey').text().split('').join('.');
      const win_rate = parseFloat(row.find('td.win').text());
      const pick_rate = parseFloat(row.find('td.popularity').text());

      return {
        skills: skill_order,
        win: win_rate,
        pick: pick_rate
      };
    });
  }
}).get();

skill_selections = R.reject(R.isNil, R.flatten(skill_selections));
```